packages/mobile/src/web3/contracts.ts

```diff
@@ -27,7 +32,18 @@ function getWeb3(fornoMode: boolean): Web3 {
       fornoMode ? DEFAULT_FORNO_URL : 'geth'
     }`
   )
-  return fornoMode ? new Web3(getHttpProvider(DEFAULT_FORNO_URL)) : new Web3(getIpcProvider())
+  return fornoMode ? new Web3(getHttpProvider(DEFAULT_FORNO_URL)) : new Web3(ipcProvider)
+}
+
+export const getWalletAsync = async () => promisifyGenerator(getWallet())
+
+export function* getWallet() {
```

Comment:
To keep consistency with our other functions like this, maybe `getConnectedWallet` would be more accurate here